addons/project/models/project.py

```diff
@@ -313,7 +313,7 @@ def _read_group_stage_ids(self, stages, domain, order):
     active = fields.Boolean(default=True,
         help="If the active field is set to False, it will allow you to hide the project without removing it.")
     sequence = fields.Integer(default=10)
-    partner_id = fields.Many2one('res.partner', string='Customer', auto_join=True, tracking=True, domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]")
+    partner_id = fields.Many2one('res.partner', string='Customer', auto_join=True, tracking=True, check_company=True, domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]")
```

Comment:
I am wondering if the domain is necessary? 🤔 

If we check that: https://github.com/odoo/odoo/blob/0bc3ae0ed58c030d32d094a3549b2c56fb1ddb6b/odoo/fields.py#L2840

it seems the domain is automatically added. 🤔 

What do you think? 🙂